### PR TITLE
Reorganize code to avoid duplication when creating a squashfs FS

### LINKED ISSUE #3973: Reorganize code to easily create squashfs file systems
## Version of Singularity:

All.

### Expected behavior

Have all the required interfaces to easily create squashfs file systems, including for testing, without code duplication.

### Actual behavior

The code to figure out the location of the squashfs binary is not public and therefore cannot be reused outside of the initial package.

### Steps to reproduce behavior

From `pkg/build/assemblers/assembler_sif.go`:
```
func getMksquashfsPath() (string, error) {
```

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

Reorganize code to avoid duplication when creating a squashfs file system.
I will use it to write unit tests for the encryption code (issue #3971).
I did not write a unit test because the function has both the looking up of the default Singularity configuration file and the extraction of the path to the `mksquashfs` binary.

**This fixes or addresses the following GitHub issues:**

- Fixes #3973

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by ikaneshiro — CHANGES_REQUESTED** [review]
This looks good. Can we move this to `internal/pkg/until/squashfs` or something like that so it is not importable to external projects. The heavy reliance on our build config makes this not very useful.

**Review by ikaneshiro — APPROVED** [review]

File: internal/pkg/build/assemblers/assembler_sif.go [hunk 1]
```diff
@@ -11,20 +11,18 @@ import (
...
 	"os/exec"
-	"path/filepath"
 	"regexp"
 	"runtime"
 	"strconv"
 	"strings"
 	"syscall"
 
+	squashfs "github.com/sylabs/singularity/internal/pkg/util/fs/squashfs"
+
```
Comment (WestleyK) [inline internal/pkg/build/assemblers/assembler_sif.go]: Extra newline here

```suggestion
```

File: internal/pkg/util/fs/squashfs/squashfs.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package squashfs
```
Comment (WestleyK) [inline internal/pkg/util/fs/squashfs/squashfs.go:6]: Need that copyright thing here

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
// This software is licensed under a 3-clause BSD license. Please consult the
// LICENSE.md file distributed with the sources of this project regarding your
// rights to use or distribute this software.

package squashfs
```

File: internal/pkg/util/fs/squashfs/squashfs.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	singularityConfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
+)
+
+// GetPath figures out where the mksquashfs binary is
+// and return an error is not available or not usable.
+func GetPath() (string, error) {
+	// Parse singularity configuration file
+	c := &singularityConfig.FileConfig{}
+	if err := config.Parser(buildcfg.SYSCONFDIR+"/singularity/singularity.conf", c); err != nil {
+		return "", fmt.Errorf("Unable to parse singularity.conf file: %s", err)
```
Comment (WestleyK) [inline internal/pkg/util/fs/squashfs/squashfs.go]:
```suggestion
		return "", fmt.Errorf("unable to parse singularity.conf file: %s", err)
```

File: internal/pkg/util/fs/squashfs/squashfs.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config"
+	singularityConfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
+)
+
+// GetPath figures out where the mksquashfs binary is
+// and return an error is not available or not usable.
+func GetPath() (string, error) {
+	// Parse singularity configuration file
+	c := &singularityConfig.FileConfig{}
+	if err := config.Parser(buildcfg.SYSCONFDIR+"/singularity/singularity.conf", c); err != nil {
```
Comment (WestleyK) [inline internal/pkg/util/fs/squashfs/squashfs.go]: Maybe use `filepath.Join()` here?

File: internal/pkg/util/fs/squashfs/squashfs.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/config"
+	singularityConfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
```
Comment (WestleyK) [inline internal/pkg/util/fs/squashfs/squashfs.go]: Probably better not to use caps in package names

```suggestion
	singularityconfig "github.com/sylabs/singularity/pkg/runtime/engines/singularity/config"
```

**Review by WestleyK — CHANGES_REQUESTED** [review]
Looks good to me, just a couple suggestions :+1:

File: internal/pkg/util/fs/squashfs/squashfs.go (hunk 4 above)
Comment (gvallee) [reply internal/pkg/util/fs/squashfs/squashfs.go]: The code base is full of this problem so I created a new internal function to get the path and created issue #3976 to fix it everywhere.

File: internal/pkg/util/fs/squashfs/squashfs.go (hunk 5 above)
Comment (gvallee) [reply internal/pkg/util/fs/squashfs/squashfs.go]: fixed

**Review by WestleyK — APPROVED** [review]
Looks good to me! Thanks :+1:

**Pull request merged**